src/ApplicationLogs/LogReader.cs

```diff
@@ -385,26 +384,27 @@ private JObject TransactionToJObject(UInt256 txHash)
 
         private JObject BlockToJObject(UInt256 blockHash)
         {
-            var blockOnPersist = _neostore.GetBlockLog(blockHash, TriggerType.OnPersist);
-            var blockPostPersist = _neostore.GetBlockLog(blockHash, TriggerType.PostPersist);
-
-            if (blockOnPersist == null && blockPostPersist == null)
-                return null;
-
             var blockJson = new JObject();
             blockJson["blockhash"] = blockHash.ToString();
             var triggerList = new List<JObject>();
 
+            var blockOnPersist = _neostore.GetBlockLog(blockHash, TriggerType.OnPersist);
             if (blockOnPersist != null)
                 triggerList.Add(BlockItemToJObject(blockOnPersist));
+
+            var blockPostPersist = _neostore.GetBlockLog(blockHash, TriggerType.PostPersist);
             if (blockPostPersist != null)
                 triggerList.Add(BlockItemToJObject(blockPostPersist));
 
             blockJson["executions"] = triggerList.ToArray();
+
+            if (blockOnPersist == null && blockPostPersist == null)
+                return null;
```

Comment:
Why create the `JObject` to return null later? I think it's more efficient the previous version

Reply:
Reason is caused by a caching issue in `LevelDb` since we are not using the right `API` calls to the `libleveldb.dll`. The old way will overwrite the previous; in this case its `_neostore.GetBlockLog(blockHash, TriggerType.OnPersist);` results that will be overwritten with the next `API` call `_neostore.GetBlockLog(blockHash, TriggerType.PostPersist);`